### PR TITLE
[5-57] Assign test post integration

### PR DESCRIPTION
**Pull request by KuznetsovR**
Post request on test assignment + data validation added

### CODE REVIEW

**Review by HenadzV — APPROVED** [review]

File: exalearn-ui/src/app/pages/assign-test-page/assign-test-page.component.ts [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 })
 export class AssignTestPageComponent implements OnInit {
-	constructor(private tableService: TableService) {}
+	constructor(private apiService: ApiService) {}
 
 	isDataAvailable = false;
 	data: UserBack[] = [];
 	ngOnInit(): void {
-		this.tableService.getData('/api/users').then((data: UserBack[]) => {
+		this.apiService.getRequest('/api/users').then((data: UserBack[]) => {
```
Comment (HrebenYury) [inline exalearn-ui/src/app/pages/assign-test-page/assign-test-page.component.ts]: Please, use constant from constants/api.constants.ts

File: exalearn-ui/src/app/components/assign-test-modal/assign-test-modal.component.html [hunk 2]
```diff
@@ -3,15 +3,26 @@ <h1 mat-dialog-title>{{ 'VIEW_MODAL.ASSIGN_TEST_ON' | translate }} {{data.firstN
...
+				[min]="minDate"
+				[matDatepicker]="picker"
+				(dateChange)="onDateChange($event)"
+				[formControl]="dateControl"
+				required
+			/>
 			<mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
 			<mat-datepicker #picker></mat-datepicker>
+			<mat-error *ngIf="dateControl.hasError('required') || !checkDateValidity()">
+				Please enter a valid date (mm/dd/yyyy)
```
Comment (HrebenYury) [inline exalearn-ui/src/app/components/assign-test-modal/assign-test-modal.component.html]: Translate it

Comment (KuznetsovR) [reply exalearn-ui/src/app/components/assign-test-modal/assign-test-modal.component.html]: translated and added to locales

**Review by HrebenYury — APPROVED** [review]

**Review by HrebenYury — APPROVED** [review]

**Review by KotdenGit — APPROVED** [review]

File: exalearn-ui/src/app/components/assign-test-modal/assign-test-modal.component.ts [hunk 3]
```diff
@@ -5,32 +5,47 @@ import { MatSelectChange } from '@angular/material/select';
...
+		public userService: UserService
 	) {
 		const currentDate = new Date();
 		this.minDate = new Date(currentDate.valueOf() + this.DayInMilliseconds);
 	}
 
 	assign(): void {
-		this.dialogRef.close();
+		if (this.level !== null && this.date !== null && this.data) {
+			this.apiService.postRequest('/api/users/assignedTest', {
```
Comment (boburwebdev) [inline exalearn-ui/src/app/components/assign-test-modal/assign-test-modal.component.ts]: You should use api.constants in postRequest

**Review by vstrabko — APPROVED** [review]

File: exalearn-ui/src/app/components/assign-test-modal/assign-test-modal.component.ts (hunk 3 above)
Comment (KuznetsovR) [reply exalearn-ui/src/app/components/assign-test-modal/assign-test-modal.component.ts]: POST_USER_ASSIGN_TEST added and used

**Pull request merged**